src/data_model.rs

```diff
@@ -0,0 +1,105 @@
+/// The size of a type.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+#[allow(missing_docs)]
+pub struct Size {
+    /// The size in bits of this type.
+    bits: u8,
+    /// The number of bits in a byte.
+    char_bit: u8,
+}
+
+impl Size {
+    /// Return the number of bits this `Size` represents.
+    pub fn bits(self) -> u8 {
+        self.bits
+    }
+
+    /// Return the number of bytes in a size.
+    ///
+    /// Depending on the target, a byte may not necessarily be 8 bits.
+    pub fn bytes(self) -> u8 {
+        let bytes = self.bits / self.char_bit;
+        // make sure that self.bits is a multiple of char_bit
+        assert_eq!(bytes * self.char_bit, self.bits);
+        bytes
+    }
+    /// Create a new `Size`, assuming that `char_bit` is 8.
+    fn with_bits(bits: u8) -> Self {
+        Self {
+            bits,
+            char_bit: 8,
+        }
+    }
+}
+
+/// The C data model used on a target.
+///
+/// See also https://en.cppreference.com/w/c/language/arithmetic_types
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+#[non_exhaustive]
+pub enum CDataModel {
+    /// The data model used most commonly on Win16. `int` and `long` are 32 bits.
```

Comment:
Is this comment correct? I'm not familiar with `LP32` specifically, but if `int` is 32 bits, why isn't this `ILP32`?

Reply:
This comment is incorrect, `int` is 16 bit on LP32.